File: src/zxh/examples/cpp/driver_cat.cpp
#include "zxhsim/zxh.h"
#include "zxhsim/runtime.h"

using ZXHSim::PrintRes;
using ZXHSim::ZXH;

int main(int argc, char **argv)
{
    ZXHSim::init(&argc, &argv);

    const int n = 1000;
    const int shots = 1000;

    {
        ZXH sim(n);
        sim.H(0);
        for (int i = 1; i < n; i++)
            sim.CX(0, i);

        sim.execute();
        auto res = sim.Sampling(shots);
        PrintRes(res, n);
    }

    ZXHSim::finalize();
    return 0;
}

File: src/zxh/examples/python/driver_cat.py
import zxhsim

n = 1000
shots = 1000

zxhsim.init()

sim = zxhsim.ZXH(n)
sim.H(0)
for i in range(1, n):
    sim.CX(0, i)

sim.execute()
res = sim.Sampling(shots)
zxhsim.PrintRes(res, n)

zxhsim.finalize()
